cf/data/data.py

```diff
@@ -10082,30 +10082,57 @@ def swapaxes(self, axis0, axis1, inplace=False, i=False):
         d._set_dask(dx, reset_mask_hardness=False)
         return d
 
-    def fits_in_memory(self, itemsize):
-        """Return True if the master array is small enough to be
-        retained in memory.
+    def fits_in_memory(self):
+        """Return True if the array is small enough to be retained in
+        memory.
+
+        Returns True if the size of the array with all delayed
+        operations computed, including a space for a full boolean
+        mask, is small enough to be retained in available memory.
+
+        The available memory is the actual free memory less the amount
+        reserved as a temporary work space, i.e. ``cf.free_memory() -
+        cf.fm_threshold()``
+
+        The delayed operations are actually not computed by
+        `fits_in_memory`, so it is possible that an intermediate
+        operation may exceed the available memory, even if the final
+        array does not.
+
+        .. seealso:: `array`, `compute`, `persist`, `cf.free_memory`,
+                     `cf.fm_threshold`
 
         :Parameters:
 
-            itemsize: `int`
+            itemsize: deprecated at version TODODASK
                 The number of bytes per word of the master data array.
 
         :Returns:
 
             `bool`
+                Whether or not the computed array fits in memory.
 
         **Examples**
 
-        >>> print(d.fits_in_memory(8))
+        >>> d = cf.Data(9, 'm')
+        >>> d.fits_in_memory()
+        True
+
+        >>> size = int(2 * cf.free_memory() / 8)
```

Comment:
It took me a while to clock that the insistence in dividing by eight (in the test as well as here) instead of simplifying to `/4` for these cases is to account for (as a guess that might be wrong?) the typical amount of bytes required per float for a `numpy.array` data structure, so perhaps a comment to clarify the relevance could be helpful to users? And/or there's the `nbytes` attribute which  you could call for these examples to show explicitly that the memory exceeds the available?

Reply:
Thanks, Sadie - I've hopefully covered all of the good things you raise in https://github.com/NCAS-CMS/cf-python/pull/401/commits/a763e7d25a9a2c2a5f84ffc46e5f9ca70004c1dd

On reflection I've also removed the `fm_threshold` component. Given that dask essentially manages its own memory, I think that there's no point in us trying to micromanage it, too.

Reply:
Brilliant, all good thanks (and from this I noticed the "always including space for a full boolean mask" description which I missed before, apologies, which clarifies the situation at hand further!).

> On reflection I've also removed the fm_threshold component. Given that dask essentially manages its own memory, I think that there's no point in us trying to micromanage it, too.

Hear, hear!

I've sanity-checked that new commit and all feedback is addressed and everything is good to go, so marking this as resolved. Please merge.